commit aa1ee6e0898d177e6bcf49237d82c804bc410985
Author: Karim Belabas <Karim.Belabas@math.u-bordeaux.fr>
Date:   Tue Nov 27 15:40:16 2018 +0100

    47- gmp kernel: off-by-1 error in red_mongomery
    
    [read uninitialized memory, then ignores result]

diff --git a/src/kernel/gmp/mp.c b/src/kernel/gmp/mp.c
index 1c30d0a..3a534f5 100644
--- a/src/kernel/gmp/mp.c
+++ b/src/kernel/gmp/mp.c
@@ -1005,7 +1005,7 @@ red_montgomery(GEN T, GEN N, ulong inv)
   }
   if (carry)
   { /* Td > N overflows (k+1 words), set Td := Td - N */
-    GEN NE = N + k+2;
+    GEN NE = N + k+1;
     Td = Te;
     Nd = Ne;
     t = subll(*++Td, *++Nd); *Td = t;
